src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,218 @@
+/-
+Copyright (c) 2022 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import linear_algebra.finite_dimensional
+
+/-!
+
+# Projective Spaces
+
+This file contains the definition of the projectivization of a vector space over a field,
+as well as the bijection between said projectivization and the collection of all one
+dimensional subspaces of the vector space.
+
+## Constructing terms of `projectivization K V`.
+We have three ways to construct terms of `projectivization K V`:
+- `projectivization.mk K v hv` where `v : V` and `hv : v ≠ 0`.
+- `projectivization.mk' K v` where `v : { w : V // w ≠ 0 }`.
+- `projectivization.mk'' H h` where `H : submodule K V` and `h : finrank H = 1`.
+
+## Other definitions
+- For `v : projectivization K V`, `v.submodule` gives the corresponding submodule of `V`.
+- `projectivization.equiv_submodule` is the equivalence between `projectivization K V`
+  and `{ H : submodule K V // finrank H = 1 }`.
+- For `v : projectivization K V`, `v.rep : V` is a representative of `v`.
+
+## Projects
+Everything in this file can be done for `division_ring`s instead of `field`s, but
+this would require a significant refactor of the results from
+`linear_algebra.finite_dimensional` and its imports.
+
+-/
+
+variables (K V : Type*) [field K] [add_comm_group V] [module K V]
+
+/-- The setoid whose quotient is the projectivization of `V`. -/
+def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
+(mul_action.orbit_rel Kˣ V).comap coe
+
+/-- The projectivization of the `K`-vector space `V`. -/
+def projectivization := quotient (projectivization_setoid K V)
+
+namespace projectivization
+
+variables {V}
+
+/-- Construct an element of the projectivization from a nonzero vector. -/
+def mk (v : V) (hv : v ≠ 0) : projectivization K V := quotient.mk' ⟨v,hv⟩
+
+/-- A variant of `projectivization.mk` in terms of a subtype. `mk` is preferred. -/
+def mk' (v : { v : V // v ≠ 0 }) : projectivization K V := quotient.mk' v
+
+@[simp] lemma mk'_eq_mk (v : { v : V // v ≠ 0}) :
+  mk' K v = mk K v v.2 :=
+by { dsimp [mk, mk'], congr' 1, simp }
+
+noncomputable instance [nontrivial V] : inhabited (projectivization K V) :=
+let e := exists_ne (0 : V) in ⟨mk K e.some e.some_spec⟩
```

Comment:
I don't know how common `noncomputable` `inhabited` instances are, but they feel a bit pointless - I'd lean towards
```lean
instance [nontrivial V] : nonempty (projectivization K V) :=
let ⟨v, hv⟩ := exists_ne (0 : V) in ⟨mk K v hv⟩
```
and just silence the `inhabited` linter.

Reply:
Done